ratchet-tree.go

```diff
@@ -252,7 +296,8 @@ func (t *RatchetTree) Decap(from leafIndex, context []byte, path *DirectPath) []
 		panic(fmt.Errorf("mls.rtn:Decap Malformed leaf node"))
 	}
 
-	t.Nodes[from].merge(&RatchetTreeNode{PublicKey: &path.Nodes[0].PublicKey})
+	leafNode := toNodeIndex(from)
```

Comment:
@suhasHere - This was the problematic line that was causing test failures.

Reply:
ah! how did i miss this !!! cool catch :-)